src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,100 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents the duke.Duke chat-bot, called Nite.
+ * Nite is an interactive app which helps to keep track of tasks.
+ * @author Chia Wen Ling
+ * @version v0.1
+ */
+public class Duke {
+    private static Ui ui;
+    private static Storage storage;
+    private static TaskList tasks;
+    private boolean isInitialised;
+
+    /**
+     * Creates a duke.Duke Chat-bot.
+     *
+     * @param filePath Directory where duke.Duke text file is saved.
+     * @param folderPath Path name of duke.Duke text file to be saved.
+     */
+    public Duke(String filePath, String folderPath) {
+        ui = new Ui();
+        storage = new Storage(filePath, folderPath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public Duke() {
+
+    }
+
+    /**
+     * Runs the chat-bot upon starting the program.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command c = Parser.parse(fullCommand);
```

Comment:
'c' is not a very intuitive variable name. if i rmb correctly, single character names should only be used for scratch variables